airflow/contrib/operators/gcp_spanner_operator.py

```diff
@@ -197,3 +197,192 @@ def execute(self, context):
+                                              database_id=self.database_id,
+                                              ddl_statements=self.ddl_statements)
+        else:
+            self.log.info("The database '{}' in project '{}' and instance '{}'"
+                          " already exists. Not modifying it.".
+                          format(self.database_id, self.project_id, self.instance_id))
+        return True
+
+
+class CloudSpannerInstanceDatabaseUpdateOperator(BaseOperator):
+    """
+    Updates a Cloud Spanner database with DDL statement specified.
+
+    :param project_id: The ID of the project which owns the instances, tables and data.
+    :type project_id: str
+    :param instance_id: Cloud Spanner instance ID.
+    :type instance_id: str
+    :param database_id: Cloud Spanner database ID.
+    :type database_id: str
+    :param ddl_statements: String list containing DDL to apply to the database.
+    :type ddl_statements: [str]
+    :param gcp_conn_id: The connection ID used to connect to Google Cloud Platform.
+    :type gcp_conn_id: str
+    """
+    # [START gcp_spanner_database_update_template_fields]
+    template_fields = ('project_id', 'instance_id', 'database_id', 'ddl_statements',
+                       'gcp_conn_id')
+    template_ext = ('.sql', )
+    # [END gcp_spanner_database_update_template_fields]
+
+    @apply_defaults
+    def __init__(self,
+                 project_id,
+                 instance_id,
+                 database_id,
+                 ddl_statements,
+                 gcp_conn_id='google_cloud_default',
+                 *args, **kwargs):
+        # type: (str, str, str, [str], str, object, object) -> None
+        self.instance_id = instance_id
+        self.project_id = project_id
+        self.database_id = database_id
+        self.ddl_statements = ddl_statements
+        self.gcp_conn_id = gcp_conn_id
+        self._validate_inputs()
+        self._hook = CloudSpannerHook(gcp_conn_id=gcp_conn_id)
+        super(CloudSpannerInstanceDatabaseUpdateOperator, self).__init__(*args, **kwargs)
+
+    def _validate_inputs(self):
+        if not self.project_id:
+            raise AirflowException("The required parameter 'project_id' is empty")
+        if not self.instance_id:
+            raise AirflowException("The required parameter 'instance_id' is empty")
+        if not self.database_id:
+            raise AirflowException("The required parameter 'database_id' is empty")
+        if not self.ddl_statements:
+            raise AirflowException("The required parameter 'ddl_statements' is empty")
+
+    def execute(self, context):
+        if not self._hook.get_database(self.project_id,
+                                       self.instance_id,
+                                       self.database_id):
+            raise AirflowException("The Cloud Spanner database "
+                                   "'{}' in project '{}' and instance '{}' is missing".
```

Comment:
Maybe add some more "actionable" log message? Something like:
`... . Create the database first before you can update it.`

I'm not sure if it's necessary, though.

Reply:
Added. Nice touch.